tool/tctl/common/autoupdate_command.go

```diff
@@ -0,0 +1,217 @@
+	c.getCmd.Flag("proxy", `URL of the proxy`).StringVar(&c.proxy)
+
+	c.watchCmd = clientToolsCmd.Command("watch", "Start monitoring auto update target version updates.")
+	c.watchCmd.Flag("proxy", `URL of the proxy`).StringVar(&c.proxy)
+
+	if c.stdout == nil {
+		c.stdout = os.Stdout
+	}
+}
+
+// TryRun takes the CLI command as an argument and executes it.
+func (c *AutoUpdateCommand) TryRun(ctx context.Context, cmd string, client *authclient.Client) (match bool, err error) {
+	switch cmd {
+	case c.updateCmd.FullCommand():
+		err = c.Upsert(ctx, client)
+	case c.getCmd.FullCommand():
+		err = c.Get(ctx, client)
+	case c.watchCmd.FullCommand():
+		err = c.Watch(ctx, client)
+	default:
+		return false, nil
+	}
+	return true, trace.Wrap(err)
+}
+
+// Upsert works with AutoUpdateConfig and AutoUpdateVersion resources to create or update.
+func (c *AutoUpdateCommand) Upsert(ctx context.Context, client *authclient.Client) error {
+	if c.mode != "" {
+		config, err := client.GetAutoUpdateConfig(ctx)
+		if trace.IsNotFound(err) {
+			if config, err = autoupdate.NewAutoUpdateConfig(&autoupdatev1pb.AutoUpdateConfigSpec{}); err != nil {
+				return trace.Wrap(err)
+			}
+		} else if err != nil {
+			return trace.Wrap(err)
+		}
+		if config.Spec.Tools == nil {
+			config.Spec.Tools = &autoupdatev1pb.AutoUpdateConfigSpecTools{}
+		}
+		if config.Spec.Tools.Mode != c.mode {
+			config.Spec.Tools.Mode = c.mode
+			if _, err := client.UpsertAutoUpdateConfig(ctx, config); err != nil {
+				return trace.Wrap(err)
+			}
+			fmt.Fprint(c.stdout, "autoupdate_config has been upserted\n")
+		}
+	}
+
+	version, err := client.GetAutoUpdateVersion(ctx)
+	if trace.IsNotFound(err) {
+		if version, err = autoupdate.NewAutoUpdateVersion(&autoupdatev1pb.AutoUpdateVersionSpec{}); err != nil {
+			return trace.Wrap(err)
+		}
+	} else if err != nil {
+		return trace.Wrap(err)
+	}
+	if version.Spec.Tools == nil {
+		version.Spec.Tools = &autoupdatev1pb.AutoUpdateVersionSpecTools{}
+	}
+	if version.Spec.Tools.TargetVersion != c.toolsTargetVersion {
+		version.Spec.Tools.TargetVersion = c.toolsTargetVersion
+		if _, err := client.UpsertAutoUpdateVersion(ctx, version); err != nil {
+			return trace.Wrap(err)
+		}
+		fmt.Fprint(c.stdout, "autoupdate_version has been upserted\n")
+	}
+
+	return nil
+}
+
+// Get makes request to fetch tools autoupdate version, if proxy flag is not set
+// authorized handler should be used.
+func (c *AutoUpdateCommand) Get(ctx context.Context, client *authclient.Client) error {
+	response, err := c.get(ctx, client)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := utils.WriteJSON(c.stdout, response); err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+// Watch launch the watcher of the tools auto update target version updates to pull the version
+// every minute.
+func (c *AutoUpdateCommand) Watch(ctx context.Context, client *authclient.Client) error {
```

Comment:
What's the goal of the watcher here? 

if the desired version === teleport.SemVersion, nothing will be printed which won't give useful information

Reply:
there were no strict requirement for this one, I going to change to empty version by default, it might be useful not to print version if the current version is the same as target version if we do autoupdate manually instead of integrated in {tctl,tsh} (otherwise we should execute command `version` then compare it with that we get from `watch` command)